### PR TITLE
test(cardano-services): fixes a not deterministic error in tests

### PR DESCRIPTION
**Pull request by iccicci**
# Context

Our unit tests are fetching `http://some_url/unfetchable` to check error handling.
The problem is that the error may depend on external system (the dns server) causing the error to be not deterministic: this can make the tests to fail.

# Proposed Solution

Changed the URL to call into a mocked URL responding with error.

### CODE REVIEW

**Review by IvayloAndonov — APPROVED** [review]
LGTM!

File: packages/cardano-services/test/StakePool/HttpStakePoolMetadataService/HttpMetadataService.test.ts [hunk 1]
```diff
@@ -202,23 +203,19 @@ describe('StakePoolMetadataService', () => {
...
-        };
+        return { body: 'not found', code: 404 };
       };
 
       ({ closeMock, serverUrl } = await mockPoolExtMetadataServer(handler));
 
       metadata = {
         ...mainExtMetadataMock,
         extDataUrl: `${serverUrl}/${ExtMetadataFormat.CIP6}`,
-        extSigUrl: UNFETCHABLE,
```
Comment (mkazlauskas) [inline packages/cardano-services/test/StakePool/HttpStakePoolMetadataService/HttpMetadataService.test.ts:221]: Is this variable still needed? If yes, maybe update other tests that use it as well?

**Review by mkazlauskas — APPROVED** [review]

File: packages/cardano-services/test/StakePool/HttpStakePoolMetadataService/HttpMetadataService.test.ts (hunk 1 above)
Comment (iccicci) [reply packages/cardano-services/test/StakePool/HttpStakePoolMetadataService/HttpMetadataService.test.ts:221]: It is used in other tests which do not perform the fetch.
I tried to do that but that would increase the test complexity as now a pre computed hash can be used (the URL is always the same), changing it requires computing the hash on some more tests... I think this doesn't worth.

Comment (mkazlauskas) [reply packages/cardano-services/test/StakePool/HttpStakePoolMetadataService/HttpMetadataService.test.ts:221]: Alright, let's keep it

**Pull request merged**